src/ui/screens/SubIdentities/AddSubIdentity.test.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import {render, fireEvent} from 'src/testUtils';
+import {AddSubIdentity} from './AddSubIdentity';
+
+jest.useFakeTimers();
+
+const bottomSheetModal = {
+  dismiss: jest.fn(),
+};
+
+const addSubIdentity = {
+  subIdentity: jest.fn(),
+};
+
+describe('', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should render the AddSubIdentity component', async () => {
+    const {findByText, findByPlaceholderText} = render(
+      <AddSubIdentity onClose={bottomSheetModal.dismiss} onAddPress={addSubIdentity.subIdentity} />,
+    );
+    await findByPlaceholderText('Sub account name (optional)');
+    await findByPlaceholderText('👉 Paste address here, e.g. 167r...14h');
+    await findByText('Cancel');
+    await findByText('Add Identity');
+  });
+
+  it('should close the bottom-sheets when pressed on cancel button', async () => {
+    const bottomSheetsSpy = jest.spyOn(bottomSheetModal, 'dismiss');
+    const {findByText, findByTestId} = render(
+      <AddSubIdentity onClose={bottomSheetModal.dismiss} onAddPress={addSubIdentity.subIdentity} />,
+    );
+    await findByText('Cancel');
+    fireEvent.press(await findByTestId('cancel-identity-button'));
+    expect(bottomSheetsSpy).toBeCalledTimes(1);
+  });
+
+  it('should add new sub-identity', async () => {
```

Comment:
How about:
` it('should call onAddPress with the right arguments when the form is valid and Add Identity button is pressed', () => {...})`

We are not actually testing that the sub-identity was added to something